app/conversations/document_based.py

```diff
@@ -22,8 +22,8 @@ def __init__(self):
         """
 
         self.llm = OobaboogaLLM()
-        self.text_splitter = CharacterTextSplitter(
-            chunk_size=1000, chunk_overlap=0)
+        self.text_splitter = RecursiveCharacterTextSplitter(
+            chunk_size=600, chunk_overlap=20, length_function=len)
```

Comment:
As a rule of thumb I tend to make sure that the chunk_size - chunk_overlap is not superior to the seq_length of the model (512 for instructor-xl) so we're sure we're not losing data

Reply:
limited it to 500.